src/index.ts

```diff
@@ -69,17 +71,24 @@ async function retryFetch(
   jsonFetchOptions: JsonFetchOptions,
 ): Promise<Response> {
   const shouldRetry = jsonFetchOptions.shouldRetry ?? DEFAULT_SHOULD_RETRY;
+  const onRequestStart =
+    typeof jsonFetchOptions.onRequestStart === 'function' ? jsonFetchOptions.onRequestStart : null;
+  const onRequestEnd =
+    typeof jsonFetchOptions.onRequestEnd === 'function' ? jsonFetchOptions.onRequestEnd : null;
   const retryOptions = {...DEFAULT_RETRY_OPTIONS, ...jsonFetchOptions.retry};
   const requestOptions = getRequestOptions(jsonFetchOptions);
 
   try {
     const response = await promiseRetry(async (throwRetryError, retryCount) => {
+      if (onRequestStart !== null) onRequestStart(requestUrl, retryCount);
       try {
         const res = await fetch(requestUrl, requestOptions);
         if (shouldRetry(res)) throwRetryError(null);
+        if (onRequestEnd !== null) onRequestEnd(res, retryCount);
```

Comment:
don't these two lines need to be swapped? otherwise we'll never reach the `onRequestEnd`, right?

Reply:
Good catch, thank you

Reply:
@serhalp although i agree with you here, this change is causing to fail the test I have created where `onRequestStart` should be called 6 times, but with this change is being duplicated, 12 instead of 6, I'm still trying to figure out, let me know if you see a possible cause for this.

Reply:
BTW if I leave the call `onRequestEnd` in same line where it is currently, the test pass, is confusing because calls of `onRequestEnd` should not affect the times that `onRequestStart` is called, supposedly 🤦🏻‍♂️